cache/surrogate/specification.md

```diff
@@ -90,7 +112,16 @@ URL to. The server will store the resource URL in each provided surrogate key.
 
   My awesome content
   ~~~
-The server **MUST** store the URL resource inside the `my-key` and `my-second-key` surrogate-keys.
+The server **MUST** store the URL resource inside the `my-key` and `my-second-key` surrogate-keys.  
+For each key not already stored for this resource, the server **SHOULD** return the directive `stored`.
```

Comment:
What is the intended use for this?

I assume `server` here is not the `Server` definition you define early on, as it seems more relevant to `Cache` definition?

Actual Server and Client have little interest in this information. If a Surrogate Key is not stored/applied to a cache entry, then it would simply be omitted no?

As mentioned in my previous review, [Fastly provides debugging via `Fastly-Debug: 1` request header](https://developer.fastly.com/reference/http-headers/Surrogate-Key/#debugging), which retains the `Surrogate-Key` header on responses returned.

If you need more information beyond that, you might provide that info via a different endpoint or logging?

Reply:
With that, client know which tags are linked to the resource.

Reply:
Who is the client here? A web browser is a type of client right? Why does that need to know what tags are associated to a cached resource?

Only the cache proxy really needs to know this in most interactions. This is why providers like Fastly strip the header from the response to clients, unless you enable the debug header for example.

Reply:
The client is the one defined in the RFC, not always a web browser (https://datatracker.ietf.org/doc/html/rfc1856#section-2.1). A web browser is one implementation of a possible client.

Your cache **can** be placed as proxy between your proxy (e.g. Fastly) and your application.

Reply:
> not always a web browser

I understand this, I said "A web browser is a type of client right?".

I still don't see a reason that justifies providing internal information to any client by default. Fastly has approached this correctly by stripping the header unless opt-in debug request header is included IMO.

Reply:
> With that, the client knows which tags are linked to the resource.

> Your cache **can** be placed between your proxy (e.g. Fastly) and your application.

Please explain why the client needs to know this information.

Is it a specific type of client? A Web browser client or mobile app client aren't likely to need this information are they?

Reply:
RFC redaction dropped.